src/include/mpir_typerep.h

```diff
@@ -81,4 +81,23 @@ int MPIR_Typerep_unpack_external(const void *inbuf, void *outbuf, MPI_Aint outco
 
 void MPIR_Typerep_debug(MPI_Datatype type);
 
+int MPIR_Typerep_pack_reduce(const void *inbuf, MPI_Aint incount, MPI_Datatype datatype,
+                             MPI_Aint inoffset, void *outbuf, MPI_Aint max_pack_bytes,
+                             MPI_Aint * actual_pack_bytes, MPI_Op op);
+int MPIR_Typerep_unpack_reduce(const void *inbuf, MPI_Aint insize,
+                               void *outbuf, MPI_Aint outcount, MPI_Datatype datatype,
+                               MPI_Aint outoffset, MPI_Aint * actual_unpack_bytes, MPI_Op op);
+int MPIR_Typerep_reduce_is_supported(MPI_Op op, MPI_Datatype datatype);
+
+int MPIR_Typerep_ipack_reduce(const void *inbuf, MPI_Aint incount, MPI_Datatype datatype,
+                              MPI_Aint inoffset, void *outbuf, MPI_Aint max_pack_bytes,
+                              MPI_Aint * actual_pack_bytes, MPIR_Typerep_req * typereq_req,
+                              MPI_Op op);
+int MPIR_Typerep_iunpack_reduce(const void *inbuf, MPI_Aint insize, void *outbuf, MPI_Aint outcount,
+                                MPI_Datatype datatype, MPI_Aint outoffset,
+                                MPI_Aint * actual_unpack_bytes, MPIR_Typerep_req * typereq_req,
+                                MPI_Op op);
+int MPIR_Typerep_op(void *source_buf, int source_count, MPI_Datatype source_dtp,
+                    void *target_buf, int target_count, MPI_Datatype target_dtp,
+                    MPI_Op acc_op, int src_kind);
```

Comment:
Let's only add the API needed for this PR.

Reply:
I mean, you should only expose the function that is needed by RMA. All other functions should be `static` internal functions. We can easily expose more functions when they are needed.

Reply:
This is done @hzhou

Reply:
I'll take another review later but it looks good.